src/shared_modules/indexer_connector/src/monitoring.hpp

```diff
@@ -86,78 +86,91 @@ class Monitoring final
     }
 
     /**
-     * @brief Class constructor. Initializes values map and check servers' health.
+     * @brief Class constructor. Checks the servers' health.
      *
-     * @param values Servers to be monitored.
+     * @param serverAddresses Servers to be monitored.
      * @param interval Interval for monitoring.
      * @param secureCommunication Object that provides secure communication.
      */
-    explicit Monitoring(const std::vector<std::string>& values,
+    explicit Monitoring(const std::vector<std::string>& serverAddresses,
                         const uint32_t interval = INTERVAL,
                         const SecureCommunication& secureCommunication = {})
         : m_interval(interval)
     {
-        // Initialize the map with the values, all servers are available.
-        for (auto& value : values)
+        /**
+         * @brief Health check function
+         *
+         * @note It sends a request to the \p serverAddress and update the \p serverStatus (true if the server is
+         * available, false otherwise).
+         */
+        const auto healthCheck = [this, secureCommunication](const std::string& serverAddress, bool& serverStatus)
         {
-            m_values[value] = true;
+            // Get the health of the server.
+            HTTPRequest::instance().get(
+                HttpURL(serverAddress + "/_cat/health?v"),
+                [&](std::string response)
+                {
+                    // Remove the tabs and double spaces.
+                    Utils::replaceAll(response, "\t", " ");
+                    response = Utils::trimRepeated(response, ' ');
+                    // Split the response by rows.
+                    const auto rows {Utils::split(response, '\n')};
+
+                    // Check if the response has the expected number of rows.
```

Comment:
```suggestion
                    serverStatus = false;
```

Reply:
I don't fully undertand this request

Reply:
this assignation is to avoid the else statement and reduce the complexity/length of this method/std::function